### PR TITLE
remove auth token validation from iot grpc ingest

### PR DESCRIPTION
**Pull request by jeffgrunewald**
the grpc streams are coming over plaintext and signed by the submitting gateway so there's not much value in trying to roll out a "secret" bearer token for authentication purposes to restrict submission of reports to the ingest bucket for the iot/lorawan network. additionally, there is utility in being able to filter inputs from sources, such as avoiding network "crosstalk" of mainnet reports being submitted to testing network buckets and vice versa, but there are better, more ergonomic mechanisms for providing validation against these types of erroneous submissions so we should just remove the restriction and simplify the ingest code path.

### CODE REVIEW

**Review by madninja — COMMENTED** [review]
1. Remove api token from env default if any 
2. Add network env entry, default to mainnet 
3. Check keytype before checking signature of received reports. Drop if keytype not the same as the configured network for the ingester

**Comment by andymck** [comment]
lg, but do we need to apply same changes to the 5g flow?

File: ingest/src/server_lora.rs [hunk 1]
```diff
@@ -18,6 +18,37 @@ pub type GrpcResult<T> = std::result::Result<Response<T>, Status>;
...
+        lora_witness_report_tx: file_sink::MessageSender,
+    ) -> Result<Self> {
+        Ok(Self {
+            lora_beacon_report_tx,
+            lora_witness_report_tx,
+            required_network: Self::required_network()?,
+        })
+    }
+
+    fn verify_network(&self, public_key: &PublicKey) -> std::result::Result<(), Status> {
```
Comment (madninja) [inline ingest/src/server_lora.rs:36]: Use GrpcResult?

**Comment by madninja** [comment]
> lg, but do we need to apply same changes to the 5g flow?

Yes please

File: ingest/src/server_lora.rs [hunk 2]
```diff
@@ -3,21 +3,52 @@ use chrono::Utc;
...
+        } else {
+            Err(Status::invalid_argument("invalid network"))
+        }
+    }
+
+    fn required_network() -> Result<Network> {
+        env::var("REQUIRED_NETWORK")
+            .map_or_else(
+                |_| Ok(Network::MainNet),
+                |value| Network::from_str(value.as_str()),
```
Comment (madninja) [inline ingest/src/server_lora.rs:48]: you can't use `&value` here?

File: ingest/src/server_lora.rs (hunk 1 above)
Comment (bbalser) [reply ingest/src/server_lora.rs]: Fixed c55a08a

File: ingest/src/server_lora.rs (hunk 2 above)
Comment (bbalser) [reply ingest/src/server_lora.rs]: Fixed c55a08a

File: ingest/src/server_lora.rs [hunk 3]
```diff
@@ -29,26 +29,17 @@ impl GrpcServer {
...
-            required_network: Self::required_network()?,
+            required_network: required_network()?,
         })
     }
 
-    fn verify_network(&self, public_key: &PublicKey) -> std::result::Result<(), Status> {
+    fn verify_network(&self, public_key: &PublicKey) -> GrpcResult<()> {
         if self.required_network == public_key.network {
-            Ok(())
+            Ok(Response::new(()))
```
[the remark this answers is not in the document]
Comment (jeffgrunewald) [reply ingest/src/server_lora.rs:38]: nope; it unwraps the `Ok` and gets `^^ expected struct `tonic::Response`, found `()``

[the remark this answers is not in the document]
Comment (madninja) [reply ingest/src/server_lora.rs:38]: ugh.. bah even with `().into()`eh?

[the remark this answers is not in the document]
Comment (jeffgrunewald) [reply ingest/src/server_lora.rs:38]: tonic result doesn't implement `From` for either `()` or `Result<(), _>`

[the remark this answers is not in the document]
Comment (bbalser) [reply ingest/src/server_lora.rs:38]: If I change that back to Ok(()), it doesn't compile
```
 ^^ expected struct `tonic::Response`, found `()
 ```

**Review by madninja — APPROVED** [review]

**Pull request merged**